pennylane/ops/functions/equal.py

```diff
@@ -72,12 +73,10 @@ def equal(
         >>> qml.equal(op3, op4, check_trainability=False)
         True
     """
+    if isinstance(op1, (Adjoint, Controlled, Exp, Pow, Prod, SProd, Sum)):
```

Comment:
Why not just use `op1.arithmetic_depth  > 0` depth check below? Then we don't have to maintain this list of current arithmetic operators.

Reply:
Nice catch!